### PR TITLE
[Merged by Bors] - fix(validator_client): return http 404 rather than 405 in http api

### LINKED ISSUE #4596: Return 404 instead of 405 in VC HTTP API
## Description

We fixed https://github.com/sigp/lighthouse/issues/3112 in the beacon node API but the same issue affects the validator client API. See that issue and the BN fix for details: https://github.com/sigp/lighthouse/pull/3836

An example of a request that triggers the error:

```fish
curl -s -H "Authorization: Bearer "(cat token.txt) "http://localhost:5062/not_found" | jq
```

```json
{
  "code": 405,
  "message": "METHOD_NOT_ALLOWED",
  "stacktraces": []
}
```

### PR DESCRIPTION
**Pull request by duguorong009**
## Issue Addressed
- Close #4596 

## Proposed Changes
- Add `Filter::recover` to handle rejections specifically as 404 NOT FOUND

Please list or describe the changes introduced by this PR.

## Additional Info

Similar to PR #3836

### CODE REVIEW

File: validator_client/src/http_api/mod.rs [hunk 1]
```diff
@@ -1187,15 +1188,18 @@ pub fn serve<T: 'static + SlotClock + Clone, E: EthSpec>(
                         .or(post_fee_recipient)
                         .or(post_gas_limit)
                         .or(post_std_keystores)
-                        .or(post_std_remotekeys),
+                        .or(post_std_remotekeys)
+                        .recover(warp_utils::reject::handle_rejection),
                 ))
-                .or(warp::patch().and(patch_validators))
+                .or(warp::patch()
+                    .and(patch_validators.recover(warp_utils::reject::handle_rejection)))
```
Comment (michaelsproul) [inline validator_client/src/http_api/mod.rs:1195]:
```suggestion
                    .and(patch_validators).recover(warp_utils::reject::handle_rejection))
```

**Comment by michaelsproul** [comment]
Looks good. ~~I think you might just need to run `cargo fmt --workspace`~~

Lets see if CI passes. Then I'll do some more manual testing, and we can merge.

**Comment by duguorong009** [comment]
> Looks good. ~I think you might just need to run `cargo fmt --workspace`~
> 
> Lets see if CI passes. Then I'll do some more manual testing, and we can merge.

FYI, I ran `cargo fmt --all` for code.
But, there is no change at all.

Hmm, there is CI failure. I am not much sure why.
Can you let me know what I can do with CI error? @michaelsproul

**Comment by michaelsproul** [comment]
We can ignore that failure, it's just a flakiness in the tests. I'll double-check with some manual testing then we can merge

**Comment by duguorong009** [comment]
@michaelsproul @chong-he 
Um, can you let me know what stops the merge?
If anything I should do, pls let me know.

**Comment by chong-he** [comment]
> @michaelsproul @chong-he Um, can you let me know what stops the merge? If anything I should do, pls let me know.

It seems ok to me: https://github.com/sigp/lighthouse/pull/4758/commits/0ba06afb4e2bd6c182edf658617b3be9e758f8da

Do you see any error? 

As a side note I will be doing some testing on this and hope to get back to you soon

**Comment by chong-he** [comment]
Tested both GET and POST HTTP and it is all good. I think we are good to merge this @michaelsproul 

Details:

GET:

```bash
curl -X GET "http://localhost:5062/lighthouse/testing" -H "Authorization: Bearer api-token-0x02d34c396c87ddcc0690e3f6b8f7fe23fdf5f309f8569fe9147dc3bf5f445db587" | jq
```

POST:

```bash
DATADIR=/var/lib/lighthouse
curl -X POST http://localhost:5062/lighthouse/testing \
-H "Authorization: Bearer $(cat ${DATADIR}/validators/api-token.txt)" \
-H "Content-Type: application/json" \
-d '[
    {
        "enable": true,
        "description": "validator_one",
        "deposit_gwei": "32000000000",
        "graffiti": "Mr F was here",
        "suggested_fee_recipient": "0xa2e334e71511686bcfe38bb3ee1ad8f6babcc03d"
    },
    {
        "enable": false,
        "description": "validator two",
        "deposit_gwei": "34000000000"
    }
]' | jq
```

With `Lighthouse v4.5.0-441fc16` both returned:

```json
{
  "code": 405,
  "message": "METHOD_NOT_ALLOWED",
  "stacktraces": []
}
```

With this PR `Lighthouse v4.5.0-0ba06af` both returned:

```json
{
  "code": 404,
  "message": "NOT_FOUND",
  "stacktraces": []
}
```

**Review by michaelsproul — COMMENTED** [review]
Great, lets merge this.

Sorry for the delay, we had a merge-freeze in effect while v4.5.0 was happening.

**Review by michaelsproul — APPROVED** [review]

**Pull request closed without merging**